src/mqttcm_connect.c

```diff
@@ -427,28 +426,35 @@ void on_connect(struct mosquitto *mosq, void *obj, int reason_code, int flag, co
 	MqttCMInfo("on_connect: success. broker_connect set to 1\n");
 	broker_connect = 1;
 
+	//For Mqtt reconnection case, use in-memory subscribe list to subscribe the components again
 	if(reconnectFlag)
 	{
-		while(1)
+		//printList();
+		comp_topic_name_t* temp = g_head;
+
+		while (temp != NULL)
 		{
-			int subscribeSuccessFlag = 1;
+			MqttCMDebug("Inside mqtt subscribe of %s\n", temp->compName);
+			mqtt_subscribe(temp->compName, temp->topic);
+			temp = temp->next;
+		}
+
+		reconnectFlag = 0;
+	}
+	else  //For crash or process restart case to get the subscribe details from file
+	{
+		if(GetTopicFromFileandUpdateList())
```

Comment:
Do we have an info print with compName and topic when added to file? This will help to debug if any issues in process restart cases

Reply:
yes it is added in AddTopicToFile while updating the file